### DISCUSSION TITLE
proposal for new functionality

### DISCUSSION THREAD

**Discussion by axlcorp** [simpleavr/s14clock#4] [Ideas]
Hello, 
I have just purchased 2 x S14Clock LED Word Clock Kit (24 Characters) .
and I would like to propose you a list of ideas for new features for the software part.

Software : 
- language management for the web interface and the display of the time in letters. 
- add a menu to manage NTP servers to have the choice of public and/or private server (I have an ntp server at home)
- display of the weather (example : https://github.com/chubin/wttr.in)
- add a Rest APi to control the display of text for example via scripts.
- change the SSID name from esp32ap to esp32ap-xxxx with xxxx a random number (I have two S14clock and I had a SSID conflict) 

Hardware : 
- I would like to add 'a DHT11 or DHT22 to have the humidity and temperature of the room.

here are the few ideas I have in mind and for my last question could you tell me if your code is under open source license ? 

thank you in advance for your feedback, and a big thank you to you for what you have created it's really great

**Comment by simpleavr** [comment]
Thanks for your input;

Software :

- language management for the web interface and the display of the time in letters.
**not sure if I can manage this as I only speak 2 languages and my English is half-good.**
- add a menu to manage NTP servers to have the choice of public and/or private server (I have an ntp server at home)
**this is quite possible, although I would prefer less configuration, and this could reside in an "advance" menu.**
- display of the weather (example : https://github.com/chubin/wttr.in)
**not really sure which options to choose, most I found requires registration and key, etc. could push into the "advance" settings.**
- add a Rest APi to control the display of text for example via scripts.
**right now it got a similar interface, say the restful api is called "msg", you can push messages like "192.169.2.xxx/msg/HELLO_WORLD via the "msg" node.**
- change the SSID name from esp32ap to esp32ap-xxxx with xxxx a random number (I have two S14clock and I had a SSID conflict)
**I will look into that. I don't access the softAP after initial setup though. And access the autoConnect menu via the station mode connection (when device joined your home network).**

**I will continue responding (hardware and others) later in another entry.**

**Comment by simpleavr** [comment]
As for hardware, right now the design had used up all exposed 27 IO pins and so even when the DHT11 requires only 1 pin, it cannot accommodate. The 27 pins used are 14 for segments, 8 for 1st 8 digits, 5 for the last 16 digits via the 74HC154D 4-to-16 line decoder.

The hardware design when I started this project 1+ years ago employs the Wemos D1 mini, with various LED and other drivers. I had reached this simplest (for now) design by adopting the S2 mini and employing multiplexing scheme. There were reliability problems in the older design where there are many components including smd devices that can be easily be messed up during soldering.

It is only possible with the current simpler design that I can offer this as a kit.

I am in the process of further simplifying the design to eliminate the 74HC154D, the design will incorporate some charliplexing (on top of multiplexing) to reduce pin count, I reckon it is possible to use 26 (14+12) IO pins. In that case, we might be able to spare one pin to communicate the DHT11. I might expose some this extra pin, V+, Gnd for such purpuse.

This repository is open source, please feel free to branch, clone or do whatever you like with the source code. I can put a lot of time into this project as I have other interests, so I don't intend to open the repository to contributors.

My focus for the firmware is to provide stability to S14 Clock owners, will include bug fixes, new features will only be introduced if they are in demand, simple to use, and can work reliability on current and future hardware.

With the software is available freely, advance users could implement new functions and features. If they are features of general interests, I can look at them and incorporate them into the main releases. But right now I am just working on the following items;

- Solving the mystery of sometimes the SoftAP won't accept password '12345678'.
- Simplifying the hardware design and eliminate the 74HC154D decoder.
- Additional transaction state (eye candy).
